### ISSUE TITLE
`Mix.install/2` doesn't work if the previous call failed

### ISSUE THREAD

**Issue by dannote** [elixir-lang/elixir#10957]
### Environment

* Erlang/OTP 24 [RELEASE CANDIDATE 2]
* Elixir 1.12.0-rc.0
* Operating system: macOS 11.3

### Current behavior

Let's try to install a package with incorrect version supplied:

```elixir
iex(1)> Mix.install([{:timex, "~ 3.7"}])
** (Mix.Error) Required version "~ 3.7" for package timex is incorrectly specified (from: mix.exs)
    (mix 1.12.0-rc.0) lib/mix.ex:454: Mix.raise/2
    (hex 0.21.2) lib/hex/remote_converger.ex:189: Hex.RemoteConverger.verify_package_req/4
    (elixir 1.12.0-rc.0) lib/enum.ex:930: Enum."-each/2-lists^foreach/1-0-"/2
    (hex 0.21.2) lib/hex/remote_converger.ex:172: Hex.RemoteConverger.verify_input/2
    (hex 0.21.2) lib/hex/remote_converger.ex:42: Hex.RemoteConverger.converge/2
    (mix 1.12.0-rc.0) lib/mix/dep/converger.ex:95: Mix.Dep.Converger.all/4
    (mix 1.12.0-rc.0) lib/mix/dep/converger.ex:51: Mix.Dep.Converger.converge/4
    (mix 1.12.0-rc.0) lib/mix/dep/fetcher.ex:16: Mix.Dep.Fetcher.all/3
```

That worked as expected, but then any correct installation attempts fail:

```elixir
iex(1)> Mix.install([{:timex, "~> 3.7"}])
** (Mix.Error) Mix.install/2 cannot be used inside a Mix project
    (mix 1.12.0-rc.0) lib/mix.ex:454: Mix.raise/2
    (mix 1.12.0-rc.0) lib/mix.ex:516: Mix.install/2
```

After restarting `IEx` everything works as expected:

```elixir
iex(1)> Mix.install([{:timex, "~> 3.7"}])
:ok
```

### Expected behavior

Mix should be able to install packages after such mistakes.

**Comment by Eiji7** [comment]
Same here
```elixir
Erlang/OTP 24 [RELEASE CANDIDATE 3] [erts-12.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Interactive Elixir (1.12.0-rc.0) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)>
```

**Comment by bmitc** [comment]
I'm getting the same error in a LiveBook notebook.

![image](https://user-images.githubusercontent.com/65685447/152624622-1e1ea4aa-6c4f-41d6-a92d-0a6b7d6aa4cb.png)

**Comment by josevalim** [comment]
Correct. If you upgrade your Livebook, it will show a Reconnect button for you to make it easy to restart.

**Comment by bmitc** [comment]
Hi Jose. Yes, I've seen the restart VM button bellow the cell (unless you are referring to another button). :) I should have provided more details or maybe filed another issue. Apologies for that!

In this case, the restart button had appeared once before. I then removed the versioning from the dependency and got this error again (there was some other error in between, I think due to a syntax issue like leaving the `{}` around the dependency). That time, the restart button didn't appear, and the only way to recover was to restart the LiveBook server. This is on a fairly recent version of LiveBook (I installed it within the past week or two) and Elixir 1.13.
